### ISSUE TITLE
JSON fields doesn't work in the Boolean field.

### ISSUE THREAD

**Issue by dkulyk** [laravel/nova-issues#598]
Nova 1.017

Required to fix #596 

If use 'field->option' attributes Field doesn't resolve value in the Boolean field.

[Reproducible code example](https://github.com/dkulyk/nova-issues/tree/json_fields_2)

**Comment by victorpfm** [comment]
It doesn't work in Nova 1.0.19 either. If I use a Text field, it displays true, but Boolean displays always false.

**Comment by alexbowers** [comment]
Have you tested this against the latest version?

**Comment by dkulyk** [comment]
The v1.1.3 has a the bug

**Comment by ragingdave** [comment]
Based on the code example I believe you want the cast type to be object, not json. Based on the framework code, json and array are the same, and are then treated like an array.

**Comment by dkulyk** [comment]
Exactly.

> Based on the code example I believe you want the cast type to be object, not json. Based on the framework code, json and array are the same, and are then treated like an array.

**Comment by ragingdave** [comment]
So....you have your app configured incorrectly.....and it's not a bug?

**Comment by dkulyk** [comment]
Why? JSON - is a text format. Exactly i mean type [casting to array](https://github.com/laravel/framework/blob/5.7/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L489-L491). 
Actually this bug fixed in the #596 . But in the Boolean field resolving the value is incorrect:
```php
class Field
{
    protected function resolveAttribute($resource, $attribute)
    {
        return data_get($resource, str_replace('->', '.', $attribute));
    }
}
```
But Boolean override this method
```php
class Boolean extends Field
{
    protected function resolveAttribute($resource, $attribute)
    {
        return data_get($resource, $attribute) == $this->trueValue ? true : false;
    }
}
```

**Comment by nickels** [comment]
on my model I have:

```
    protected $casts = [
        'options' => 'array'
    ];
```

In the nova resource I'm trying to have a Boolean field:

```
Laravel\Nova\Fields\Boolean::make('Club Members Only', 'options.clubmembers'),
```

It renders correctly, however can't update the value to true or false.

Data is stored in the db using a `jsonb` field like so:

`{"clubmembers": "1"}`

Thoughts?

**Comment by tobz-nz** [comment]
I'm having the same issue - with text/number fields. data is displayed correctly but there's a js error when trying to save.

`Unhandled promise rejection TypeError: "e.t0.response is undefined; can't access its "status" property"`

**Comment by bonzai** [comment]
This has been fixed and will be included in the next release.

**Comment by StanAngeloff** [comment]
Confirmed solved in v1.2.1 Orion released December 12, 2018.
